java-squid/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java

```diff
@@ -287,6 +289,15 @@ private void visit(Tree tree, Tree terminator) {
           programState = ProgramState.put(programState, variableTree.symbol(), unstack.b.get(0));
         }
         break;
+      case TYPE_CAST:
+        TypeCastTree typeCast = (TypeCastTree) tree;
+        Type type = typeCast.type().symbolType();
+        if (type.isPrimitive()) {
+          Pair<ProgramState, List<SymbolicValue>> unstack = ProgramState.unstack(programState, 1);
+          programState = unstack.a;
+          programState = ProgramState.stackValue(programState, constraintManager.createSymbolicValue(typeCast.expression()));
+        }
+        break;
       case ASSIGNMENT:
         AssignmentExpressionTree assignmentExpressionTree = ((AssignmentExpressionTree) tree);
         // FIXME restricted to identifiers for now.
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1134)